src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -98,6 +98,21 @@ public static StringComparer Create(CultureInfo culture, bool ignoreCase)
             return new CultureAwareComparer(culture, ignoreCase);
         }
 
+        public static StringComparer Create(CultureInfo culture, CompareOptions options)
+        {
+            if (culture == null)
+            {
+                throw new ArgumentException(nameof(culture));
+            }
+
+            if ((options & CultureAwareComparer.ValidCompareMaskOffFlags) != 0)
+            {
+                throw new ArgumentException(SR.Argument_InvalidFlag, nameof(options));
+            }
```

Comment:
onen last comment: you may move this check to the CultureAwareComparer constructor so you don't have to use ValidCompareMaskOffFlags here and everything will be well contained inside CultureAwareComparer